checkov/terraform/plan_parser.py

```diff
@@ -125,15 +125,13 @@ def _find_child_modules(child_modules: ListNode) -> List[Dict[str, Dict[str, Any
     return resource_blocks
 
 
-def parse_tf_plan(tf_plan_file: str, out_parsing_errors: Dict[str, str]) -> Tuple[Optional[Dict[str, Dict[str, Any]]], Optional[List[Tuple[int, str]]]]:
+def parse_tf_plan(tf_defintions: Dict[str, Dict[str, Any]], template_lines: Dict[str, List[Tuple[int, str]]], tf_plan_file: str, out_parsing_errors: Dict[str, str]):
```

Comment:
@YaaraVerner why does this method now accepts `tf_definitions` and not returning it? `create_definitions` anyways return the definitions, so I don't see a reason to add additional arguments, and It's not a best practice to modify in-place a function argument, unless we have a good reason for it

Reply:
until now for each plan file, we parsed it and run the checks on it. the function is parsing only one file and I wanted to save all the results in the same object. but I agree that it isn't the best solution, I will change it and extend the tf_definitions outside the function.

Reply:
@tronxd done :)